src/blockchain/mod.rs

```diff
@@ -68,10 +69,10 @@ pub struct KvStoreChain<K: KvStore> {
 }
 
 impl<K: KvStore> KvStoreChain<K> {
-    pub fn new(kv_store: K) -> Result<KvStoreChain<K>, BlockchainError> {
+    pub fn new(kv_store: K, starting_difficulty: Option<u32>) -> Result<KvStoreChain<K>, BlockchainError> {
```

Comment:
I think it would be better to pass the gensis_block itself instead of a starting_difficulty